src/utils/hooks/useFetchInstitutions.ts

```diff
@@ -1,43 +1,38 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useCallback } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { useTranslation } from 'react-i18next';
-import Axios from 'axios';
 
 import { InstitutionUnitBase } from '../../types/institution.types';
 import { getInstitutions } from '../../api/institutionApi';
 import { setNotification } from '../../redux/actions/notificationActions';
 import { NotificationVariant } from '../../types/notification.types';
 import { RootStore } from '../../redux/reducers/rootReducer';
 import { setInstitutions } from '../../redux/actions/institutionActions';
+import useCancelToken from './useCancelToken';
 
 // This hook is used to fetch all top-level institutions
 const useFetchInstitutions = (): [InstitutionUnitBase[], boolean] => {
   const dispatch = useDispatch();
   const { t } = useTranslation('feedback');
   const institutions = useSelector((store: RootStore) => store.institutions);
   const [isLoading, setIsLoading] = useState(true);
+  const cancelToken = useCancelToken();
 
-  useEffect(() => {
-    const cancelSource = Axios.CancelToken.source();
-
-    const fetchInstitutions = async () => {
-      const response = await getInstitutions(cancelSource.token);
-      if (response) {
-        if (response.error) {
-          dispatch(setNotification(t('error.get_institutions'), NotificationVariant.Error));
-        } else if (response.data) {
-          dispatch(setInstitutions(response.data));
-        }
-        setIsLoading(false);
+  const fetchInstitutions = useCallback(async () => {
+    const response = await getInstitutions(cancelToken);
+    if (response) {
+      if (response.error) {
+        dispatch(setNotification(t('error.get_institutions'), NotificationVariant.Error));
+      } else if (response.data) {
+        dispatch(setInstitutions(response.data));
       }
-    };
-    // Institutions should not change, so ensure we fetch only once
-    if (!institutions || institutions.length === 0) {
-      fetchInstitutions();
+      setIsLoading(false);
     }
+  }, [dispatch, t, cancelToken]);
 
-    return () => cancelSource.cancel();
-  }, [t, dispatch, institutions]);
+  useEffect(() => {
+    fetchInstitutions();
+  }, [fetchInstitutions]);
 
```

Comment:
nå hentes institusjoner hver gang man åpner dropdown, men hvis man har hentet tidligere så vil den være populert på forhånd, så det vil ikke merke det (med mindre man har byttet språk siden sist)
det betyr at det vil bli en del unødvendige requester, men finner ingen kjempegod fiks på det :/

Reply:
vi kunne evt lagt inn språk i redux også
```
institutions: { language: "nob", items: [...] }
```
da kunne vi hentet igjen hvis språk er endret siden sist

Reply:
men vi har vel språk i localstorage? kan ikke det brukes? hvis ikke, så stemmer jeg for å putte det i Redux ja. Det tar jo ikke akkurat så mye plass 😅

Reply:
vi har det i LS ja, men vi må vite hvilket språk institusjonene vi har er hentet på. hvis den er hentet på norsk, og vi har byttet språk til engelsk etterpå må vi hente institusjoner på nytt (på engelsk)

Reply:
skjønner! kanskje like så greit å ha det i Redux da?

Reply:
det er kanskje det...